pycbc/inference/models/relbin.py

```diff
@@ -490,19 +491,31 @@ def multi_loglikelihood(self, models):
         if not hasattr(self, 'hihj'):
             self.calculate_hihjs(models)
 
-        # finally add in the lognl term from this model
-        for m1, m2 in itertools.combinations(models, 2):
-            for det in self.data:
-                a0, a1, fedge = self.hihj[(m1, m2)][det]
-
-                fp, fc, dtc, hp, hc, h00 = m1._current_wf_parts[det]
-                fp2, fc2, dtc2, hp2, hc2, h002 = m2._current_wf_parts[det]
-
-                h1h2 = self.mlik(fedge,
-                                 fp, fc, dtc, hp, hc, h00,
-                                 fp2, fc2, dtc2, hp2, hc2, h002,
-                                 a0, a1)
-                loglr += - h1h2.real # This is -0.5 * re(<h1|h2> + <h2|h1>)
+        if self.still_needs_det_response:
+            for m1, m2 in itertools.combinations(models, 2):
+                for det in self.data:
+                    a0, a1, fedge = self.hihj[(m1, m2)][det]
+                    dtc, channel, h00 = m1._current_wf_parts[det]
+                    dtc2, channel2, h002 = m2._current_wf_parts[det]
+                    c1c2 = self.mlik(fedge,
+                                    dtc, channel, h00,
```

Comment:
Please keep the same indentation as the above line. Same as 502-503 line.

Reply:
Just to check, do you mean like this?

```
for det in self.data:
                    a0, a1, fedge = self.hihj[(m1, m2)][det]

                    dtc, channel, h00 = m1._current_wf_parts[det]
                    dtc2, channel2, h002 = m2._current_wf_parts[det]

                    c1c2 = self.mlik(fedge,
                                    dtc, channel, h00,
                                    dtc2, channel2, h002,
                                    a0, a1)
                    loglr += - c1c2.real # This is -0.5 * re(<h1|h2> + <h2|h1>)
```

Reply:
> Just to check, do you mean like this?
> 
> ```
> for det in self.data:
>         a0, a1, fedge = self.hihj[(m1, m2)][det]
>         
>         dtc, channel, h00 = m1._current_wf_parts[det]
>         dtc2, channel2, h002 = m2._current_wf_parts[det]
>         
>         c1c2 = self.mlik(fedge,
>                         dtc, channel, h00,
>                         dtc2, channel2, h002,
>                         a0, a1)
>         loglr += - c1c2.real # This is -0.5 * re(<h1|h2> + <h2|h1>)
> ```

I mean `fedge`, `dtc`, `dtc2`, and `a0` should be on the same column.

Reply:
Corrected in latest commit